web/src/test/java/org/cbioportal/web/DataAccessTokenControllerTest.java

```diff
@@ -30,69 +30,73 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-package org.cbioportal.weblegacy;
+package org.cbioportal.web;
 
-import java.net.*;
 import java.util.*;
 import org.cbioportal.model.DataAccessToken;
-import org.cbioportal.security.spring.authentication.PortalUserDetails;
 import org.cbioportal.service.DataAccessTokenService;
-import org.cbioportal.weblegacy.DataAccessTokenControllerConfig;
-import org.cbioportal.weblegacy.exception.DataAccessTokenNoUserIdentityException;
-import org.cbioportal.weblegacy.exception.DataAccessTokenProhibitedUserException;
+import org.cbioportal.service.exception.DataAccessTokenNoUserIdentityException;
+import org.cbioportal.service.exception.DataAccessTokenProhibitedUserException;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Matchers;
 import org.mockito.Mockito;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.User;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.test.context.web.WebAppConfiguration;
 
 @RunWith(SpringJUnit4ClassRunner.class)
-//@WebAppConfiguration
-@ContextConfiguration(classes = {DataAccessTokenControllerConfig.class})
+@WebAppConfiguration
+@ContextConfiguration("/applicationContext-web.xml")
```

Comment:
Ok, I think I see. We are moving the test into the web module, so we don't need to do package-scan to find the controller beans. I worry a little about bringing in the whole runtime applicationContext-web file ... potentially changes to this file in the future will break our unit tests. But maybe that is good (that problems show up in the unit tests before we commit changes to applicationContext-web)